### PR TITLE
Downsampling script for tadpoles

### PR DESCRIPTION
**Pull request by alessandrofelder**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

Keep track of how we downsampled the tadpole data.

**What does this PR do?**

Add script for tadpole downsampling and organising (to go from "sourcedata" to "rawdata")

## References

\

## How has this PR been tested?

Visual inspection of tadpole data. Downsampling functionality added with tests in #21 

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

Not yet

## Checklist:

- [ ] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: examples/tadpole/downsample_source_images.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    in_plane_resolution = 1
+    out_of_plane_resolution = 3
+    in_plane_factor = int(target_isotropic_resolution / in_plane_resolution)
+    axial_factor = int(target_isotropic_resolution / out_of_plane_resolution)
+    downsampled = downsample_anisotropic_image_stack(
+        stack, in_plane_factor=in_plane_factor, axial_factor=axial_factor
+    )
+
+    sample_id = str(folder).split("_")[1].lower()
+    channel = "blue" if str(folder).split("_")[2] == "488" else "orange"
```
Comment (niksirbi) [inline examples/tadpole/downsample_source_images.py:25]: I noticed they don't always use the absolute exact wavelength. Perhaps we should instead check if the wavelength is within ~10/20 nm of the target wavelength value for a given "color". But this is more for futureproofing, and may be left to future refactoring.

File: examples/tadpole/downsample_source_images.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    logger.info(f"{sample_filename} downsampled.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Downsample source images")
+    parser.add_argument(
+        "--source_data",
+        type=str,
+        help="Path to the source data folder",
```
Comment (niksirbi) [inline examples/tadpole/downsample_source_images.py:40]: Maybe also mention that we expect the source data folder to contain subfolders per subject, with them in-term containing the tiff.
s.

File: examples/tadpole/downsample_source_images.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        required=True,
+    )
+    parser.add_argument(
+        "--template_building_root",
+        type=str,
+        help="Path to the template-building root folder",
```
Comment (niksirbi) [inline examples/tadpole/downsample_source_images.py:46]: Maybe alse mention that the results will be written to the "rawdata" subfolder of the templat-building-root

**Review by niksirbi — APPROVED** [review]
I think this is good to go as is, but I'm slightly confused as to why certain arguments are hard-coded in the functions (e.g. `in_plane_resolution`), while others are supplied as CLI arguments (e.g. `target_isotropic resolution`).

Is the logic that some factors - like in-plane resolution - are intrinsic to the source images themselves, and hence unchanging, while the arguments supplied through the CLI can be used to run the downsampling in different ways (e.g. send results to a different path, or change the downsampling factor)?

This discussion is largely moot, since we are planning to anyway refactor and possible used a config, but I'm just curious.

**Comment by alessandrofelder** [comment]
> but I'm just curious.

I was being a minimalist, and for our current purpose we needed to downsample to different resolutions, but all our images had the same (anisotropic) resolution?

File: examples/tadpole/downsample_source_images.py (hunk 1 above)
Comment (alessandrofelder) [reply examples/tadpole/downsample_source_images.py:25]: Yea this is still tadpole-specific code anyway. Agree with the idea though.

**Pull request merged**